File: src/main/java/com/fouan/SpringEventPublisher.java
package com.fouan;

import com.fouan.events.Event;
import com.fouan.events.EventsPublisher;
import lombok.AllArgsConstructor;
import org.springframework.context.ApplicationEventPublisher;

import javax.inject.Named;

@Named
@AllArgsConstructor
final class SpringEventPublisher implements EventsPublisher {

    private final ApplicationEventPublisher applicationEventPublisher;

    @Override
    public void fire(Event<?> event) {
        applicationEventPublisher.publishEvent(event);
    }
}

File: src/main/java/com/fouan/SpringEventPublisher.kt
package com.fouan

import com.fouan.events.Event
import com.fouan.events.EventsPublisher
import org.springframework.context.ApplicationEventPublisher
import javax.inject.Named

@Named
internal class SpringEventPublisher(
    private val applicationEventPublisher: ApplicationEventPublisher
) : EventsPublisher {

    override fun fire(event: Event<*>) = applicationEventPublisher.publishEvent(event)
}
